### PR TITLE
1433 compact codec

### LINKED ISSUE #1433: Try to use compact codec when possible
### Context

Especially for the extrinsics that are already large and called frequently. It helps reduce the ecall/ocall buffer size when passing encoded extrinsics around, hopefully.

See https://substrate.stackexchange.com/questions/679/when-to-use-compact-on-extrinsic-arguments/686#686

Note we have to bump transaction version if it's changed.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by songzhouran**

### CODE REVIEW

**Comment by songzhouran** [comment]
Closes #1433

File: tee-worker/core/parentchain/indirect-calls-executor/src/executor.rs [hunk 1]
```diff
@@ -170,7 +170,11 @@ impl<
 		})??;
 
 		let root: H256 = merkle_root::<Keccak256, _>(extrinsics);
-		Ok(OpaqueCall::from_tuple(&(call, block_hash, block_number, root)))
+		let mut block_num: u32 = 0;
+		if let Ok(number) = block_number.try_into() {
+			block_num = number;
+		}
+		Ok(OpaqueCall::from_tuple(&(call, block_hash, codec::Compact(block_num), root)))
```
Comment (Kailai-Wang) [inline tee-worker/core/parentchain/indirect-calls-executor/src/executor.rs]: How about:

```suggestion
		let parentchain_block_number: ParentchainBlockNumber =
			block_number.try_into().map_err(|_| Error::ConvertParentchainBlockNumber)?;
		Ok(OpaqueCall::from_tuple(&(call, block_hash, codec::Compact(parentchain_block_number), root)))
```

**Review by Kailai-Wang — COMMENTED** [review]
Only one small thing

**Review by Kailai-Wang — COMMENTED** [review]
Only one small thing

**Review by Kailai-Wang — APPROVED** [review]

**Review by felixfaisal — APPROVED** [review]
LGTM

**Pull request merged**